api_tests/nodes/views/test_node_draft_registration_detail.py

```diff
@@ -156,9 +156,10 @@ def draft_registration(self, user, project_public, schema):
 
     @pytest.fixture()
     def reg_schema(self):
-        return RegistrationSchema.objects.get(
-            name='OSF Preregistration',
-            schema_version=SCHEMA_VERSION)
+        schema = RegistrationSchema.objects.get_latest_version(name='OSF Preregistration')
+        schema.active = True
+        schema.save()
+        return schema
```

Comment:
The old tests use a real schema too, but just a different version. I am not sure how this fixes any broken tests.

Your update uses the latest schema. One issue is that if we ever have a new version of that schema, the test will break. However, we can update the tests then.

I will leave this commit out for now and see what it breaks.

Reply:
Update: here are the two tests that using the new schema fixes.

```bash
__ TestDraftRegistrationUpdateWithNode.test_cannot_update_registration_schema __
api_tests/nodes/views/test_node_draft_registration_detail.py:408: in test_cannot_update_registration_schema
    assert res.status_code == 200
E   assert 400 == 200
E    +  where 400 = <400 Bad Request application/vnd.api+json body=b'{"errors...0"}}'/163>.status_code


_ TestDraftRegistrationUpdateWithDraftNode.test_cannot_update_registration_schema _
api_tests/nodes/views/test_node_draft_registration_detail.py:408: in test_cannot_update_registration_schema
    assert res.status_code == 200
E   assert 400 == 200
E    +  where 400 = <400 Bad Request application/vnd.api+json body=b'{"errors...0"}}'/163>.status_code
```

However, there are other places in this test that still use the old schema: https://github.com/CenterForOpenScience/osf.io/blob/cce3dd3bab8368249ab2aaf6ddcb0637915654b4/api_tests/nodes/views/test_node_draft_registration_detail.py#L546-L550, which might indicate that the old schema is OK but something else could be wrong.